src/chat/card-service.js

```diff
@@ -102,36 +61,27 @@ class CardService extends EventEmitter {
         }
     }
 
-    setRecycleBinScreen() {
-        this.archiveScreen = false;
-        this.searchScreen = false;
-        this.recycleBinScreen = true;
-        this.userScreen = false;
-        this.profileScreen = false;
-        this.uptodateScreen = false;
-        this.trigger('UptodateScreen');
-        this.trigger('SearchScreen');
-        this.trigger('RecycleBin');
-        this.trigger('ArchiveScreen');
-        this.trigger('User');
-    }
-
-    setUserScreen() {
-        this.archiveScreen = false;
-        this.searchScreen = false;
-        this.recycleBinScreen = false;
-        this.userScreen = true;
-        this.profileScreen = false;
-        this.uptodateScreen = false;
-        this.trigger('UptodateScreen');
+    setSearchScreen(content) {
+        this.searchItem = content;
         this.trigger('SearchScreen');
-        this.trigger('RecycleBin');
-        this.trigger('ArchiveScreen');
-        this.trigger('User');
     }
 
-    searchResult(content) {
-        var te = this.owner.cards;
+    setScreenType(type) {
+        this.screenType = type;
+        if (type == 1) {
```

Comment:
How about having only two lines here,

```
this.screenType = type;
this.trigger('navigation');
```
do you actually need multiple events for this? This would also simplify your event handlers; instead of binding to multiple events everywhere, you would just bind to the `navigation` event and do `this.screenType = CardService.screenType`.